### ISSUE TITLE
Package has outdated and vulnerable dependencies

### ISSUE THREAD

**Issue by rootd00d** [cgadam/artifactory-api#15]
The package is currently failing our `npm audit` scans and blocking new builds.

The following packages can be updated --
```
Package     Current  Wanted  Latest  Location                 Depended by
md5-file      5.0.0   2.0.3   5.0.0  node_modules/md5-file    artifactory-api
q             1.5.1   1.4.1   1.5.1  node_modules/q           artifactory-api
request      2.88.2  2.58.0  2.88.2  node_modules/request     artifactory-api
underscore   1.13.2   1.8.3  1.13.2  node_modules/underscore  artifactory-api
```

Below is the current status of `npm audit` --
```
# npm audit report

bl  <1.2.3
Severity: high
Remote Memory Exposure in bl - https://github.com/advisories/GHSA-pp7h-53gx-mx7r
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/bl
  request  2.16.0 - 2.80.0
  Depends on vulnerable versions of bl
  Depends on vulnerable versions of hawk
  Depends on vulnerable versions of http-signature
  Depends on vulnerable versions of qs
  Depends on vulnerable versions of tunnel-agent
  node_modules/request

cryptiles  <=4.1.1
Severity: critical
Insufficient Entropy in cryptiles - https://github.com/advisories/GHSA-rq8g-5pc5-wrhr
Depends on vulnerable versions of boom
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/cryptiles
  hawk  <=6.0.2
  Depends on vulnerable versions of boom
  Depends on vulnerable versions of cryptiles
  Depends on vulnerable versions of hoek
  Depends on vulnerable versions of sntp
  node_modules/hawk
    request  2.16.0 - 2.80.0
    Depends on vulnerable versions of bl
    Depends on vulnerable versions of hawk
    Depends on vulnerable versions of http-signature
    Depends on vulnerable versions of qs
    Depends on vulnerable versions of tunnel-agent
    node_modules/request

hawk  <=6.0.2
Severity: critical
Regular Expression Denial of Service in hawk - https://github.com/advisories/GHSA-jcpv-g9rr-qxrc
Depends on vulnerable versions of boom
Depends on vulnerable versions of cryptiles
Depends on vulnerable versions of hoek
Depends on vulnerable versions of sntp
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/hawk
  request  2.16.0 - 2.80.0
  Depends on vulnerable versions of bl
  Depends on vulnerable versions of hawk
  Depends on vulnerable versions of http-signature
  Depends on vulnerable versions of qs
  Depends on vulnerable versions of tunnel-agent
  node_modules/request

hoek  <4.2.1
Severity: moderate
Prototype Pollution in hoek - https://github.com/advisories/GHSA-jp4x-w63m-7wgm
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/hoek
  boom  <=3.1.2
  Depends on vulnerable versions of hoek
  node_modules/boom
    cryptiles  <=4.1.1
    Depends on vulnerable versions of boom
    node_modules/cryptiles
      hawk  <=6.0.2
      Depends on vulnerable versions of boom
      Depends on vulnerable versions of cryptiles
      Depends on vulnerable versions of hoek
      Depends on vulnerable versions of sntp
      node_modules/hawk
        request  2.16.0 - 2.80.0
        Depends on vulnerable versions of bl
        Depends on vulnerable versions of hawk
        Depends on vulnerable versions of http-signature
        Depends on vulnerable versions of qs
        Depends on vulnerable versions of tunnel-agent
        node_modules/request
  sntp  0.0.0 || 0.1.1 - 2.0.0
  Depends on vulnerable versions of hoek
  node_modules/sntp

http-signature  0.11.0
Severity: moderate
node-http-signature HMAC Validation Side-channel Timing Attack Information Disclosure - https://www.npmjs.com/package/http-signature,https://github.com/joyent/node-http-signature/commit/78ab1da232f31f695f5c362d863593a143aa8b56,https://github.com/joyent/node-http-signature/pull/36
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/http-signature
  request  2.16.0 - 2.80.0
  Depends on vulnerable versions of bl
  Depends on vulnerable versions of hawk
  Depends on vulnerable versions of http-signature
  Depends on vulnerable versions of qs
  Depends on vulnerable versions of tunnel-agent
  node_modules/request

qs  <6.0.4
Severity: high
Prototype Pollution Protection Bypass in qs - https://github.com/advisories/GHSA-gqgv-6jq5-jjj9
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/qs
  request  2.16.0 - 2.80.0
  Depends on vulnerable versions of bl
  Depends on vulnerable versions of hawk
  Depends on vulnerable versions of http-signature
  Depends on vulnerable versions of qs
  Depends on vulnerable versions of tunnel-agent
  node_modules/request

request  2.16.0 - 2.80.0
Severity: high
Remote Memory Exposure in request - https://github.com/advisories/GHSA-7xfp-9c55-5vqj
Depends on vulnerable versions of bl
Depends on vulnerable versions of hawk
Depends on vulnerable versions of http-signature
Depends on vulnerable versions of qs
Depends on vulnerable versions of tunnel-agent
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/request

tunnel-agent  <0.6.0
Severity: moderate
Memory Exposure in tunnel-agent - https://github.com/advisories/GHSA-xc7v-wxcw-j472
fix available via `npm audit fix --force`
Will install request@2.88.2, which is outside the stated dependency range
node_modules/tunnel-agent
  request  2.16.0 - 2.80.0
  Depends on vulnerable versions of bl
  Depends on vulnerable versions of hawk
  Depends on vulnerable versions of http-signature
  Depends on vulnerable versions of qs
  Depends on vulnerable versions of tunnel-agent
  node_modules/request

underscore  1.3.2 - 1.12.0
Severity: high
Arbitrary Code Execution in underscore - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
fix available via `npm audit fix --force`
Will install underscore@1.13.2, which is outside the stated dependency range
node_modules/underscore

11 vulnerabilities (5 moderate, 4 high, 2 critical)

To address all issues, run:
  npm audit fix --force
```

**Comment by rootd00d** [comment]
Submitting possible fix #14

**Comment by rootd00d** [comment]
Fix merged.  Thank you, @cgadam for the lightning-fast turnaround on this!
